forte/data/data_store.py

```diff
@@ -471,10 +485,18 @@ def add_link_raw(
             ``tid`` of the entry and its index in the ``type_name`` list.
 
         """
-        raise NotImplementedError
+        entry = self._new_link(type_name, parent_tid, child_tid, tid)
+        try:
+            self.__elements[type_name].add(entry)
+        except KeyError:
+            self.__elements[type_name] = SortedList(key=lambda s: (s[0], s[1]))
```

Comment:
The comparison function for `Link` can be different from the one for `Annotation`. Note that `s[0]` and `s[1]` may have different meanings for `Annotation` entry and `Link` entry.